internal/server/tlscontext/tls.go

```diff
@@ -1,4 +1,4 @@
-package server
+package tlscontext
```

Comment:
this feels a little awkward why do we need to change this?

Reply:
it was moved from the base server directory to it's own module "tlscontext" so it can be reused from both places, the agent server, and the general API server.

It's just a TLS middleware that injects the CN as context for the HTTP handlers.

Reply:
How about naming the package middleware?